expect: tcl: Can not build expect

I got the following error log
```
| checking for stdlib.h... (cached) yes
| checking for string.h... (cached) yes
| checking sys/wait.h usability... yes
| checking sys/wait.h presence... yes
| checking for sys/wait.h... yes
| checking dlfcn.h usability... yes
| checking dlfcn.h presence... yes
| checking for dlfcn.h... yes
| checking sys/param.h usability... yes
| checking sys/param.h presence... yes
| checking for sys/param.h... yes
| checking for Tcl public headers... configure: error: /work/build/tmp-glibc/work/aarch64-emlinux-linux/expect/5.45.4-r0/recipe-sysroot/usr/include/tcl8.6 directory does not contain tcl.h
| NOTE: The following config.log files may provide further information.
| NOTE: /work/build/tmp-glibc/work/aarch64-emlinux-linux/expect/5.45.4-r0/build/config.log
| ERROR: configure failed
| WARNING: exit code 1 from a shell command.
| ERROR: Function failed: do_configure (log file is located at /work/build/tmp-glibc/work/aarch64-emlinux-linux/expect/5.45.4-r0/temp/log.do_configure.5289)
ERROR: Task (/work/build/../repos/meta-debian-extended/recipes-debian/expect/expect_debian.bb:do_configure) failed with exit code '1'
NOTE: Tasks Summary: Attempted 4079 tasks of which 4078 didn't need to be rerun and 1 failed.

Summary: 1 task failed:
  /work/build/../repos/meta-debian-extended/recipes-debian/expect/expect_debian.bb:do_configure
Summary: There was 1 WARNING message shown.
Summary: There were 2 ERROR messages shown, returning a non-zero exit code.
```

The cause of this problem is that be not make /usr/include/tcl8.6/tcl.h by tcl.

current version.
```
/work/build$ find tmp-glibc/work/aarch64-emlinux-linux/tcl/8.6.9-r0/image/ -name tcl.h
tmp-glibc/work/aarch64-emlinux-linux/tcl/8.6.9-r0/image/usr/include/tcl.h
tmp-glibc/work/aarch64-emlinux-linux/tcl/8.6.9-r0/image/usr/include/tcl8.6.9/generic/tcl.h

```

in Poky
```
/work/build$ tmp-glibc/work/aarch64-emlinux-linux/tcl/8.6.9-r0/image/ -name tcl.h
/work/build/tmp-glibc/work/aarch64-emlinux-linux/tcl/8.6.9-r0/image/usr/include/tcl8.6.9/generic/tcl.h
/work/build/tmp-glibc/work/aarch64-emlinux-linux/tcl/8.6.9-r0/image/usr/include/tcl8.6/tcl.h
```

It is made changed include path by meta/recipes-devtools/tcltk/tcl/alter-includedir.patch in poky's tcl

I think that tcl shoud be generate /usr/include/tcl8.6/tcl.h.


I got the another error log after fix the above problem.
```
ERROR: expect-5.45.4-r0 do_package_qa: QA Issue: expect: The compile log indicates that host include and/or library paths were used.
         Please check the log '/work/build/tmp-glibc/work/aarch64-emlinux-linux/expect/5.45.4-r0/temp/log.do_compile' for more information. [compile-host-path]
ERROR: expect-5.45.4-r0 do_package_qa: QA run found fatal errors. Please consider fixing them.
ERROR: expect-5.45.4-r0 do_package_qa:
ERROR: expect-5.45.4-r0 do_package_qa: Function failed: do_package_qa
ERROR: Logfile of failure stored in: /work/build/tmp-glibc/work/aarch64-emlinux-linux/expect/5.45.4-r0/temp/log.do_package_qa.13212
ERROR: Task (/work/build/../repos/meta-debian-extended/recipes-debian/expect/expect_debian.bb:do_package_qa) failed with exit code '1'
NOTE: Tasks Summary: Attempted 1980 tasks of which 1979 didn't need to be rerun and 1 failed.

Summary: 1 task failed:
  /work/build/../repos/meta-debian-extended/recipes-debian/expect/expect_debian.bb:do_package_qa
Summary: There was 1 WARNING message shown.
Summary: There were 4 ERROR messages shown, returning a non-zero exit code.h
```
The cause of this problem is that be not providing appropriate include path to tcl.
and it Does not occur in poky version too.

I don't get this log when reverting debian/patches/tclprivate.diff.

--- a/recipes-debian/tcltk/tcl/alter-includedir.patch
+++ b/recipes-debian/tcltk/tcl/alter-includedir.patch
@@ -1,0 +1,80 @@
+Lets install the include header and private header files into
+usr/include/tcl8.6 when version of tcl is 8.6.x
+
+Upstream-Status: Inappropriate [Configuration Specific]
+
+Signed-off-by: Khem Raj <raj.khem@gmai.com>
+
+Fixed the TCL_INCLUDE_SPEC
+
+Signed-off-by: Robert Yang <liezhi.yang@windriver.com>
+
+Rebased for debianized Tcl 8.6.9+dfsg-2
+
+Signed-off-by: Hiraku Sugiura <hiraku.sugiura@miraclelinux.com>
+---
+ Makefile.in  | 2 +-
+ configure    | 4 ++--
+ configure.in | 4 ++--
+ 3 files changed, 5 insertions(+), 5 deletions(-)
+
+diff --git a/Makefile.in b/Makefile.in
+index ba45fae..15a5dcc 100644
+--- a/Makefile.in
++++ b/Makefile.in
+@@ -53,7 +53,7 @@ DLL_INSTALL_DIR		= @DLL_INSTALL_DIR@
+ SCRIPT_INSTALL_DIR	= $(INSTALL_ROOT)$(TCL_LIBRARY)
+ 
+ # Directory in which to install the include file tcl.h:
+-INCLUDE_INSTALL_DIR	= $(INSTALL_ROOT)$(includedir)
++INCLUDE_INSTALL_DIR	= $(INSTALL_ROOT)$(includedir)/tcl$(VERSION)
+ 
+ # Path to the private tcl header dir:
+ PRIVATE_INCLUDE_DIR	= @PRIVATE_INCLUDE_DIR@
+diff --git a/configure b/configure
+index ba41f06..01c33a3 100755
+--- a/configure
++++ b/configure
+@@ -18553,7 +18553,7 @@ eval "TCL_LIB_FILE=libtcl${LIB_SUFFIX}"
+ eval "TCL_LIB_FILE=${TCL_LIB_FILE}"
+ 
+ test -z "$TCL_LIBRARY" && TCL_LIBRARY='$(prefix)/lib/tcl$(VERSION)'
+-PRIVATE_INCLUDE_DIR='$(includedir)'
++PRIVATE_INCLUDE_DIR='$(includedir)/tcl$(VERSION)'
+ HTML_DIR='$(DISTDIR)/html'
+ 
+ # Note:  in the following variable, it's important to use the absolute
+@@ -18712,7 +18712,7 @@ TCL_BUILD_STUB_LIB_PATH="`pwd`/${TCL_STUB_LIB_FILE}"
+ TCL_STUB_LIB_PATH="${TCL_STUB_LIB_DIR}/${TCL_STUB_LIB_FILE}"
+ 
+ # Install time header dir can be set via --includedir
+-eval "TCL_INCLUDE_SPEC=\"-I${includedir}\""
++eval "TCL_INCLUDE_SPEC=\"-I${includedir}/tcl${VERSION}\""
+ 
+ #------------------------------------------------------------------------
+ # tclConfig.sh refers to this by a different name
+diff --git a/configure.in b/configure.in
+index a1c3188..71e43ce 100755
+--- a/configure.in
++++ b/configure.in
+@@ -774,7 +774,7 @@ eval "TCL_LIB_FILE=libtcl${LIB_SUFFIX}"
+ eval "TCL_LIB_FILE=${TCL_LIB_FILE}"
+ 
+ test -z "$TCL_LIBRARY" && TCL_LIBRARY='$(prefix)/lib/tcl$(VERSION)'
+-PRIVATE_INCLUDE_DIR='$(includedir)'
++PRIVATE_INCLUDE_DIR='$(includedir)/tcl$(VERSION)'
+ HTML_DIR='$(DISTDIR)/html'
+ 
+ # Note:  in the following variable, it's important to use the absolute
+@@ -895,7 +895,7 @@ TCL_BUILD_STUB_LIB_PATH="`pwd`/${TCL_STUB_LIB_FILE}"
+ TCL_STUB_LIB_PATH="${TCL_STUB_LIB_DIR}/${TCL_STUB_LIB_FILE}"
+ 
+ # Install time header dir can be set via --includedir
+-eval "TCL_INCLUDE_SPEC=\"-I${includedir}\""
++eval "TCL_INCLUDE_SPEC=\"-I${includedir}/tcl${VERSION}\""
+ 
+ #------------------------------------------------------------------------
+ # tclConfig.sh refers to this by a different name
+-- 
+2.17.1
+

--- a/recipes-debian/tcltk/tcl_debian.bb
+++ b/recipes-debian/tcltk/tcl_debian.bb
@@ -21,13 +21,14 @@ S = "${DEBIAN_UNPACK_DIR}/unix"
 DEBIAN_QUILT_DIR = "${DEBIAN_UNPACK_DIR}/.quilt_pc"
 
 FILESPATH_append = ":${COREBASE}/meta/recipes-devtools/tcltk/tcl"
-SRC_URI += "file://tcl-add-soname.patch"
+SRC_URI += "file://tcl-add-soname.patch \
+            file://alter-includedir.patch \
+           "
 SRC_URI_append_class_target += " \
            file://fix_non_native_build_issue.patch \
            file://fix_issue_with_old_distro_glibc.patch \
            file://no_packages.patch \
            file://tcl-remove-hardcoded-install-path.patch \
-           file://alter-includedir.patch \
            file://run-ptest \
 "
 

--- a/recipes-debian/tcltk/tcl_debian.bb
+++ b/recipes-debian/tcltk/tcl_debian.bb
@@ -34,6 +34,11 @@ SRC_URI_append_class_target += " \
 
 VER = "${PV}"
 
+do_debian_patch_append() {
+    cd ${S}/..
+    patch  -R -u -p1 <debian/patches/tclprivate.diff
+}
+
 inherit autotools ptest binconfig
 
 EXTRA_OECONF = "--enable-threads --disable-rpath --libdir=${libdir}"

